src/bp/core/services/middleware/event-storage.ts

```diff
@@ -0,0 +1,114 @@
+import * as sdk from 'botpress/sdk'
+import { KnexExtension } from 'common/knex'
+import { BotpressConfig } from 'core/config/botpress.config'
+import Database from 'core/database'
+import { EventRepository } from 'core/repositories'
+import { TYPES } from 'core/types'
+import { inject, injectable, tagged } from 'inversify'
+import Knex from 'knex'
+import _ from 'lodash'
+import moment from 'moment'
+import ms from 'ms'
+
+import { SessionIdFactory } from '../dialog/session/id-factory'
+
+@injectable()
+export class EventStorage {
```

Comment:
Until now I had no idea that this service worked as collector + batch insert and I was wondering why there was a storageInterval.

Maybe you could name it as `EventCollector` with a `collectionInterval`  or keep it as is but rename the config to `dumpInterval` or something

Reply:
Thanks for the name, I was running out of ideas yesterday, definitely better! I'll go with collection for other ones